src/backend/distributed/transaction/backend_data.c

```diff
@@ -780,6 +805,113 @@ MarkCitusInitiatedCoordinatorBackend(void)
 }
 
 
+/*
+ * AssignGlobalPID assigns a global process id for the current backend if
+ * it is not already assigned
+ */
+void
+AssignGlobalPID(void)
+{
+	if (MyBackendData->globalPID != 0)
+	{
+		return;
+	}
+
+	if (!IsCitusInitiatedRemoteBackend())
+	{
+		uint64 globalPID = GenerateGlobalPID();
+
+		SpinLockAcquire(&MyBackendData->mutex);
+		MyBackendData->globalPID = globalPID;
+		SpinLockRelease(&MyBackendData->mutex);
+
+		return;
+	}
+
+	uint64 globalPID = ExtractGlobalPID();
+
+	SpinLockAcquire(&MyBackendData->mutex);
+	MyBackendData->globalPID = globalPID;
+	SpinLockRelease(&MyBackendData->mutex);
+}
+
+
+/*
+ * GetGlobalPID returns the global process id of the current backend
+ */
+uint64
+GetGlobalPID(void)
+{
+	/* assign here incase it is not assigned already */
+	AssignGlobalPID();
```

Comment:
ok, as we call `AssignGlobalPID ` in `StartupCitusBackend`, calling it in other places seems redundant / confusing.


`StartupCitusBackend` guaranteed to be called for all Citus backends, hence we should not try to re-assign.

Similarly, the early return in `AssignGlobalPID` with `MyBackendData->globalPID != 0` seems redundant and error prone as well.

We should at most convert them to asserts.

(I know that was my idea, sorry for the extra iteration)